src/fdc/DiskManipulator.cc

```diff
@@ -432,13 +441,49 @@ void DiskManipulator::create(std::span<const TclObject> tokens)
 	// initialize (create partition tables and format partitions)
 	DSKDiskImage image(filename);
 	if (sizes.size() > 1) {
-		DiskImageUtils::partition(image, static_cast<std::span<const unsigned>>(sizes), bootType);
+		unsigned partitionCount = DiskImageUtils::partition(image,
+			static_cast<std::span<const unsigned>>(sizes), bootType);
+		if (partitionCount != sizes.size()) {
+			throw CommandException("Could not create all partitions; ",
+				partitionCount, " of ", sizes.size(), " created.");
+		}
 	} else {
 		// only one partition specified, don't create partition table
 		DiskImageUtils::format(image, bootType);
 	}
 }
 
+void DiskManipulator::partition(std::span<const TclObject> tokens)
+{
+	static_vector<unsigned, MAX_PARTITIONS> sizes;
+	MSXBootSectorType bootType = MSXBootSectorType::DOS2;
+
+	for (const auto& token_ : view::drop(tokens, 3)) {
+		if (auto t = parseBootSectorType(token_.getString())) {
+			bootType = *t;
+		} else if (sizes.size() >= MAX_PARTITIONS) {
+			throw CommandException("Maximum number of partitions is ", MAX_PARTITIONS);
+		} else if (size_t sectors = parseSectorSize(token_.getString()); sectors <= ~0u) {
+			sizes.push_back(narrow<unsigned>(sectors));
+		} else {
+			throw CommandException("Partition size too large.");
+		}
+	}
```

Comment:
Is it worth trying to reuse this parsing code between `create()` and `partition()`?
(Same remark here about `sectors <= ~0u`.)